sdl2/chase_in_space/chase_in_space.odin

```diff
@@ -0,0 +1,229 @@
+		}
+		// Away from other enemies
+		for _, i in game.entities {
+			if game.entities[i].type == .ENEMY && entity != &game.entities[i] {
+				dir := entity.pos - game.entities[i].pos
+				dis := math.sqrt(dir.x * dir.x + dir.y * dir.y)
+				if dis > 0 {
+					entity.pos += dir * (1. / (dis * dis)) * 0.1 * dt
+				}
+			}
+		}
+		// Shoot
+		if entity.reload_counter <= 0 {
+			append(
+				&game.entities,
+				Entity{
+					type = .PROJECTILE,
+					pos = entity.pos,
+					vel = 0.5 * dir,
+					hp = 1,
+					bullet_decay = 750,
+				},
+			)
+			entity.reload_counter = 1000
+		} else {
+			entity.reload_counter -= dt
+		}
+	case .PROJECTILE:
+		entity.pos += entity.vel * dt
+		entity.bullet_decay -= 1.0 * dt
+		if entity.bullet_decay < 0 {
+			entity.hp = 0
+		} else {
+			player := find_entity(.PLAYER, game)
+			if player == nil do return
+			if player.pos.x < entity.pos.x && entity.pos.x < player.pos.x + 10 && player.pos.y < entity.pos.y &&
+			   entity.pos.y < player.pos.y + 10 {
+				player.hp -= 1
+				fmt.printf("HIT (HP: {})\n", player.hp)
+				entity.hp = 0
+			}
+		}
+	}
+}
+
+get_time :: proc() -> f64 {
+	return f64(sdl2.GetPerformanceCounter()) * 1000 / f64(sdl2.GetPerformanceFrequency())
+}
+
+main :: proc() {
+	assert(sdl2.Init(sdl2.INIT_VIDEO) == 0, sdl2.GetErrorString())
+	defer sdl2.Quit()
+
+	window := sdl2.CreateWindow(
+		"Odin Game",
+		sdl2.WINDOWPOS_CENTERED,
+		sdl2.WINDOWPOS_CENTERED,
+		640,
+		480,
+		sdl2.WINDOW_SHOWN,
+	)
+	assert(window != nil, sdl2.GetErrorString())
+	defer sdl2.DestroyWindow(window)
+
+	// Must not do VSync because we run the tick loop on the same thread as rendering.
+	renderer := sdl2.CreateRenderer(window, -1, sdl2.RENDERER_ACCELERATED)
+	assert(renderer != nil, sdl2.GetErrorString())
+	defer sdl2.DestroyRenderer(renderer)
+
+	tickrate := 240.0
+	ticktime := 1000.0 / tickrate
+
+	game := Game {
+		renderer = renderer,
+		time     = get_time(),
+		dt       = ticktime,
+		entities = make([dynamic]Entity),
+	}
+	defer delete(game.entities)
+
+	append(&game.entities, Entity{type = .PLAYER, pos = {50.0, 400.0}, hp = 10})
+	append(&game.entities, Entity{type = .ENEMY, pos = {50.0, 50.0}, hp = 1})
+	append(&game.entities, Entity{type = .ENEMY, pos = {100.0, 100.0}, hp = 1})
+	append(&game.entities, Entity{type = .ENEMY, pos = {200.0, 200.0}, hp = 1})
+
+	dt := 0.0
+
+	for {
+		event: sdl2.Event
+		for sdl2.PollEvent(&event) > 0 {
+			#partial switch event.type {
+			case .QUIT:
+				return
+			case .KEYDOWN:
+				if event.key.keysym.scancode == sdl2.SCANCODE_ESCAPE do return
+			}
+		}
+
+		time := get_time()
+		dt += time - game.time
+
+		game.keyboard = transmute([]b8)sdl2.GetKeyboardStateAsSlice()
+		game.time = time
+
+		// Running on the same thread as rendering so in the end still limited by the rendering FPS.
+		for dt >= ticktime {
+			dt -= ticktime
+
+			for _, i in game.entities do update_entity(&game.entities[i], &game)
+
+			for i := 0; i < len(game.entities); {
```

Comment:
Here I'd probably but `/**/` in the post slot just to indicate it doesn't do anything